src/data/ProgramsD2Repository.ts

```diff
@@ -87,20 +101,109 @@ export class ProgramsD2Repository implements ProgramsRepository {
         return d2ProgramRules.run(options);
     }
 
+    async getAll(options: MoveProgramAttributeOptions): Async<ProgramAttributes[]> {
+        const trackedEntities = await this.getFromTracker<D2TrackedEntity>("trackedEntities", {
+            orgUnitIds: undefined,
+            programIds: [options.programId],
+        });
+
+        return trackedEntities.map(tei => {
+            return {
+                id: tei.trackedEntity,
+                orgUnit: tei.orgUnit,
+                trackedEntityType: tei.trackedEntityType,
+                attributes: tei.attributes.map(attribute => {
+                    return {
+                        attribute: attribute.attribute,
+                        value: attribute.value,
+                        storedBy: attribute.storedBy,
+                    };
+                }),
+                programId: options.programId,
+            };
+        });
+    }
+
+    async saveAttributes(programs: ProgramAttributes[]): Async<Stats> {
```

Comment:
For me, `ProgramAttributes` is a confusing name. It's a TrackedEntityInstance with some basic fields, including attributes. We don't a TEI entity yet. I'd create it (TrackedEntity.ts, as the new version of DHIS drop the "Instance"), and move this code to a new repository for this entity. What do you think?